pkg/k8s/resource_ctors.go

```diff
@@ -433,24 +450,24 @@ func CiliumExternalWorkloads(lc cell.Lifecycle, cs client.Clientset, opts ...fun
 	return resource.New[*cilium_api_v2.CiliumExternalWorkload](lc, lw, resource.WithMetric("CiliumExternalWorkloads")), nil
 }
 
-func CiliumEnvoyConfigResource(lc cell.Lifecycle, cs client.Clientset, opts ...func(*metav1.ListOptions)) (resource.Resource[*cilium_api_v2.CiliumEnvoyConfig], error) {
-	if !cs.IsEnabled() {
+func CiliumEnvoyConfigResource(params CiliumResourceParams, opts ...func(*metav1.ListOptions)) (resource.Resource[*cilium_api_v2.CiliumEnvoyConfig], error) {
```

Comment:
The `CiliumExternalWorkloads` resource above should be updated as well. And the same also applies to the clustermesh-apiserver specific resource in `clustermesh-apiserver/clustermesh/k8s/resources.go`. More in general, I'm a bit concerned that this is very easy to miss for new resources, but I'm not sure if there's any good solution there.

Reply:
Done. I guess having all Cilium resource constructors do this is a good start for later cut&paste!